test/extended/prometheus/prometheus.go

```diff
@@ -49,6 +53,120 @@ type TelemeterClientConfig struct {
+			"openshift-image-registry",             // https://issues.redhat.com/browse/OCPBUGS-59767
+			"openshift-operator-lifecycle-manager", // https://issues.redhat.com/browse/OCPBUGS-59768
+			"openshift-cluster-samples-operator",   // https://issues.redhat.com/browse/OCPBUGS-59769
+			"openshift-cluster-version",            // https://issues.redhat.com/browse/OCPBUGS-57585
+		)
+	)
+
+	g.BeforeEach(func(ctx g.SpecContext) {
+		var err error
+		prometheusURL, err = helper.PrometheusRouteURL(ctx, oc)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Get public url of prometheus")
+		bearerToken, err = helper.RequestPrometheusServiceAccountAPIToken(ctx, oc)
+		o.Expect(err).NotTo(o.HaveOccurred(), "Request prometheus service account API token")
+
+		if namespacesToSkip.Has(componentNamespace) {
+			e2e.Logf("The namespace %s is not skipped because $COMPONENT_NAMESPACE is set to it", componentNamespace)
+			namespacesToSkip.Delete(componentNamespace)
+		}
+	})
+
+	g.It("should not be accessible without authorization [Serial]", func() {
+		var errs []error
+
+		g.By("verifying all targets returns 401 or 403 or 500 without authorization")
+		execPod := exutil.CreateExecPodOrFail(oc.AdminKubeClient(), oc.Namespace(), "execpod-targets-authorization")
+		defer func() {
+			err := oc.AdminKubeClient().CoreV1().Pods(execPod.Namespace).Delete(context.Background(), execPod.Name, *metav1.NewDeleteOptions(1))
+			o.Expect(err).NotTo(o.HaveOccurred(), fmt.Sprintf("Delete pod %s/%s", execPod.Namespace, execPod.Name))
+		}()
+
+		contents, err := helper.GetURLWithToken(helper.MustJoinUrlPath(prometheusURL, "api/v1/targets"), bearerToken)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		targets := &prometheusTargets{}
+		err = json.Unmarshal([]byte(contents), targets)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		for _, target := range targets.Data.ActiveTargets {
+			ns := target.Labels["namespace"]
+			if !strings.HasPrefix(ns, "openshift-") {
+				continue
+			}
+			if componentNamespace != "" && ns != componentNamespace {
+				continue
+			}
+			pod := target.Labels["pod"]
+			e2e.Logf("Checking via pod exec status code from the scaple url %s for pod %s/%s without authorization (skip=%t)", target.ScrapeUrl, ns, pod, namespacesToSkip.Has(ns))
+			// Error out only when the status code is obtained
+			// Error on executing the cmd is skipped here because the endpoint does not expose any information.
+			// Promethus may have troubles to scrape it but that is out of the scape of this test
+			// The same reasoning holds for status code 500
+			err = helper.ExpectURLStatusCodeExecViaPod(execPod.Namespace, execPod.Name, target.ScrapeUrl, true, http.StatusUnauthorized, http.StatusForbidden, http.StatusInternalServerError)
+			e2e.Logf("The scaple url %s for pod %s/%s without authorization returned error %v (skip=%t)", target.ScrapeUrl, ns, pod, err, namespacesToSkip.Has(ns))
+			if err != nil && !namespacesToSkip.Has(ns) {
```

Comment:
yes, I think for readability, let's just skip ignored targets earlier, teams can always edit the list once while working on the OCPBUGS.

Reply:
I am trying to keep those.
See if https://github.com/openshift/origin/pull/30014#discussion_r2252450895 makes sense to you.
You can reply there (and I will consider this one resolved).